lib/compat/wordpress-5.9/script-loader.php

```diff
@@ -28,7 +28,7 @@ function gutenberg_enqueue_global_styles_assets() {
 		return;
 	}
 
-	$stylesheet = wp_get_global_stylesheet();
+	$stylesheet = gutenberg_get_global_stylesheet();
```

Comment:
I'm not sure whether I should be moving this entire file and [class-wp-theme-json-gutenberg.php](https://github.com/WordPress/gutenberg/pull/38816/files#diff-c220112e642d42482d91af8a96dca5ab587adfa705140148f86c064bca995631) to `compat/6.0` or leaving them here.



The[ PR description](https://github.com/WordPress/gutenberg/pull/38745) for most recent changes to this file state that "The global styles stylesheet should be loaded using this logic, no matter the WordPress version", and a comment in [load.php](https://github.com/WordPress/gutenberg/blob/0b536eb460b5c424514be962df0941f78e2c5df5/lib/load.php) says:

> [script-loader.php]  Needs to be loaded before get-global-styles-and-settings.php

If we need to make similar changes such as the one this PR is attempting, then perhaps it's worth moving both files in their entirety to 6.0 (along with these changes)?

By that I mean

```diff
diff --git a/lib/load.php b/lib/load.php
index 18d541a969..a4babbfe25 100644
--- a/lib/load.php
+++ b/lib/load.php
@@ -73,8 +73,6 @@ require __DIR__ . '/compat/wordpress-5.9/register-global-styles-cpt.php';
 // to make sure we can use the check "function_exists( 'wp_get_global_styles' )".
 // If it loads after, that function will always be present at that point
 // and the global styles assets won't be loaded.
-require __DIR__ . '/compat/wordpress-5.9/script-loader.php';
-require __DIR__ . '/compat/wordpress-5.9/get-global-styles-and-settings.php';
 require __DIR__ . '/compat/wordpress-5.9/render-svg-filters.php';
 require __DIR__ . '/compat/wordpress-5.9/json-file-decode.php';
 require __DIR__ . '/compat/wordpress-5.9/translate-settings-using-i18n-schema.php';
@@ -96,6 +94,7 @@ require __DIR__ . '/compat/wordpress-5.9/default-theme-supports.php';
 require __DIR__ . '/compat/wordpress-5.9/class-wp-rest-global-styles-controller.php';
 require __DIR__ . '/compat/wordpress-5.9/rest-active-global-styles.php';
 require __DIR__ . '/compat/wordpress-5.9/move-theme-editor-menu-item.php';
+require __DIR__ . '/compat/wordpress-6.0/script-loader.php';
 require __DIR__ . '/compat/wordpress-6.0/get-global-styles-and-settings.php';
 require __DIR__ . '/compat/wordpress-6.0/post-lock.php';
 require __DIR__ . '/compat/wordpress-6.0/blocks.php';
```

I'm not very experienced in the compat wizardry so 🤷

Reply:
Once these is rebased on top of https://github.com/WordPress/gutenberg/pull/38883 my question will be answered :)
In short, yes.